### PR TITLE
Admin create author

### PR DESCRIPTION
**Pull request by maiq721**

### CODE REVIEW

**Comment by maiq721** [comment]
ready
Link task : https://dev.framgia.com/issues/185672

File: config/routes.rb [hunk 1]
```diff
@@ -4,7 +4,12 @@
   get "/login", to: "sessions#new"
   post "/login", to: "sessions#create"
   delete "/logout", to: "sessions#destroy"
-  resources :users, only: %i(new create show)
+
+  namespace :admin do
+    resources :authors
```
Comment (mdmahmudurrahman) [inline config/routes.rb]: Specify the action

File: app/controllers/admin/authors_controller.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
+class Admin::AuthorsController < ApplicationController
+  before_action :load_author, only: %i(edit show)
+
```
Comment (mdmahmudurrahman) [inline app/controllers/admin/authors_controller.rb]: Any type of user can send request to those actions, right ?

File: config/routes.rb [hunk 3]
```diff
@@ -4,7 +4,12 @@
...
+  end
+
+  resources :users, only: %i(new create)
```
Comment (mdmahmudurrahman) [inline config/routes.rb]: I can see you have actions index, new, create, edit, update for the users

File: app/views/admin/authors/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        </td>
+        <td><%= link_to author.name, admin_author_path(author) %></td>
+        <td><%= author.brith_year %></td>
+        <td><%= author.phone %></td>
+        <td><%= author.address %></td>
+        <td>
+          <%= link_to t(".edit"), edit_admin_author_path(author), class: "btn btn-success" %>
+        </td>
+      </tr>
+    <% end %>
```
Comment (mdmahmudurrahman) [inline app/views/admin/authors/index.html.erb]: Please move this to a partial

**Comment by mdmahmudurrahman** [comment]
commented
Please add 1. input validation for the author 2. authentication 3. authorization to the controller.

**Comment by maiq721** [comment]
ready

**Comment by mdmahmudurrahman** [comment]
commented
Please fix the bug and resend again. (bow)

**Comment by maiq721** [comment]
ready 
I'm doneeee

File: app/controllers/admin/admin_controller.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+class Admin::AdminController < ApplicationController
+  
+  before_action :verify_admin
```
Comment (mdmahmudurrahman) [inline app/controllers/admin/admin_controller.rb]: It's fine you are requiring the user should be admin. One more thing you have to do is **require log in user** before a request comes to this controller.

**Comment by mdmahmudurrahman** [comment]
commented

**Comment by maiq721** [comment]
ready

File: app/controllers/admin/admin_controller.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+  def verify_admin
+  	if logged_in?
+      return if current_user.admin?
+      flash[:danger] = t ".danger"
+    else
+      redirect_to root_path
+    end
+  end
```
Comment (mdmahmudurrahman) [inline app/controllers/admin/admin_controller.rb]: Please re-write the method using the following code snippet: 
```ruby
def verify_admin 
  return if logged_in? && current_user.admin?
  flash[:danger] = "Desired message"
  redirect_to root_path
end
```

**Comment by mdmahmudurrahman** [comment]
commented. 
Please follow the comment(I think it will solve your problem related to login) and resolve the conflicts. (bow)

**Comment by huongnguyenmta** [comment]
commented

**Comment by maiq721** [comment]
ready

File: app/controllers/admin/authors_controller.rb [hunk 7]
```diff
@@ -0,0 +1,50 @@
+class Admin::AuthorsController < Admin::AdminController
+  before_action :load_author, except: %i(index new create)
+
+  def index
+    @authors = Author.all.select(:name, :phone, :address).order(name: :asc).page(params[:page]).per Settings.per_page
```
Comment (huongnguyenmta) [inline app/controllers/admin/authors_controller.rb]: `select(:name, :phone, :address).order(name: :asc)`
phần xử lý anyf tách thành 2 scope riêng trong model tương ứng nhé em

**Comment by huongnguyenmta** [comment]
commented
(dead) chú ý khi làm những pull đơn giản như này nhé em
không nên để mắc lại những lỗi đã bị comemnt từ trước nghen :-s

**Comment by maiq721** [comment]
ready
vâng ạ (bow)

**Pull request merged**